src/megachatapi_impl.cpp

```diff
@@ -5201,6 +5207,39 @@ MegaChatMessagePrivate::MegaChatMessagePrivate(const MegaChatMessage *msg)
             this->megaChatUsers->push_back(megaChatUser);
         }
     }
+
+    if (msg->getType() == TYPE_CONTAINS_META)
+    {
+        switch (msg->containsMetaType())
+        {
+            case CONTAINS_META_RICH_PREVIEW:
+            {
+                const char *text = msg->getRichPreviewText();
+                const char *title = msg->getRichPreviewTitle();
+                const char *description = msg->getRichPreviewDescription();
+                const char *image = msg->getRichPreviewImage();
+                const char *imageFormat = msg->getRichPreviewImageFormat();
+                const char *icon = msg->getRichPreviewIcon();
+                const char *iconFormat = msg->getRichPreviewIconFormat();
+                const char *url = msg->getRichPreviewUrl();
+                MegaChatRichPreview *richPreview = new MegaChatRichPreview(std::string(text), std::string(title), std::string(description),
```

Comment:
I think it's better to create a `copy()` for this class and simply call it here, instead of using all this getters, create temporal strings and so on. That's also easier to reuse in the future.

Reply:
I have follow this approach because the the parameter that we receive at this method  is a MegaChatMessage and the interface of this class only has getters to the contained info. We can make a casting to MegaChatMessagePrivate and get the MegaChatRichPreview object instead of get their fields one to one